src/bin/ion/commands/beta/generate/utils.rs

```diff
@@ -20,49 +20,144 @@ pub struct Import {
     pub(crate) type_name: String,
 }
 
-/// Represent the programming language for code generation.
-#[derive(Debug, Clone, PartialEq)]
-pub enum Language {
-    Rust,
-    Java,
+pub trait Language {
+    /// Provides a file extension based on programming language
+    fn file_extension() -> String;
+
+    /// Returns string representation of programming language
+    fn string_value() -> String;
```

Comment:
I think `Language::name` would be more descriptive? We can also allow the caller to decide whether allocation is necessary.

```suggestion
    fn name() -> &str;
```